src/sql/base/browser/ui/table/tableCellEditorFactory.ts

```diff
@@ -96,11 +98,21 @@ export class TableCellEditorFactory {
 
 			public loadValue(item: Slick.SlickData): void {
 				this._originalValue = self._options.valueGetter(item, this._args.column) ?? '';
-				this._input.value = this._originalValue;
+				if (inputType === 'date') {
+					this._input.inputElement.valueAsDate = new Date(this._originalValue);
+				} else {
+					this._input.value = this._originalValue;
+				}
 			}
 
 			public applyValue(item: Slick.SlickData, state: string): void {
 				const activeCell = this._args.grid.getActiveCell();
+				if (inputType === 'date') {
+					// Usually, the date picker will return the date in the local time zone and change the date to the previous day.
```

Comment:
I haven't checked, please share some examples of UI picked value to returned value by date picker.

Reply:
![image](https://user-images.githubusercontent.com/6816294/236324612-ce2eddf5-0fc7-4f68-98fb-2a23a10beaf6.png)

Reply:
Side note - what's that spinner in the bottom left? That shouldn't be there until you click OK, right?

Reply:
I am waiting on my other PR to get merged in before that error is fixed.

Reply:
@Charles-Gagnon but you're right the loading spinner should not show up in that situation.